src/generator/AutoRest.Go/TemplateModels/ServiceClientTemplateModel.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+using AutoRest.Core.ClientModel;
+using AutoRest.Core.Utilities;
+
+namespace AutoRest.Go.TemplateModels
+{
+    public class ServiceClientTemplateModel : ServiceClient
+    {
+        public readonly string BaseClient;
+        public string ClientName { get; set; }
+        public string ClientDocumentation { get; set; }
+        public readonly string MethodGroupName;
+        public readonly string PackageName;
+
+        public readonly MethodScopeProvider MethodScope;
+        public readonly List<MethodTemplateModel> MethodTemplateModels;
+
+        public ServiceClientTemplateModel(ServiceClient serviceClient, string packageName, string methodGroupName = null)
+        {
+            this.LoadFrom(serviceClient);
+
+            MethodGroupName = methodGroupName == null
+                                ? string.Empty
+                                : methodGroupName;
+            PackageName = packageName == null
+                            ? string.Empty
+                            : packageName;
+
+            // TODO (gosdk): Derive client name from Swagger once the Swagger settles down.
```

Comment:
Removed this comment as this is not relevant now. This was for the ManagementClient name which we have kept same in all packages.

```Go
// ManagementClient is the base client for Batch.
type ManagementClient struct {
	autorest.Client
	BaseURI        string
	APIVersion     string
	SubscriptionID string
}
```